From 352f3a90f78ddb432d844f2c08ce9761650cf33b Mon Sep 17 00:00:00 2001
From: Damjan Jovanovic <damjan@apache.org>
Date: Sun, 22 Jul 2012 13:06:16 +0000
Subject: [PATCH] Method lowerBound in
 org.apache.commons.imaging.common.itu_t4.T4AndT6Compression has a division
 error.

Jira issue key: IMAGING-88
Submitted by: Craig Kelly <craig dot n dot kelly at gmail dot com>



git-svn-id: https://svn.apache.org/repos/asf/commons/proper/imaging/trunk@1364286 13f79535-47bb-0310-9956-ffa450edef68
---
 .../commons/imaging/common/itu_t4/T4AndT6Compression.java       | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/java/org/apache/commons/imaging/common/itu_t4/T4AndT6Compression.java b/src/main/java/org/apache/commons/imaging/common/itu_t4/T4AndT6Compression.java
index 6d2bb6b66..838ac415c 100644
--- a/src/main/java/org/apache/commons/imaging/common/itu_t4/T4AndT6Compression.java
+++ b/src/main/java/org/apache/commons/imaging/common/itu_t4/T4AndT6Compression.java
@@ -720,7 +720,7 @@ private static T4_T6_Tables.Entry lowerBound(T4_T6_Tables.Entry[] entries,
         int first = 0;
         int last = entries.length - 1;
         do {
-            int middle = (first + last) >>> 2;
+            int middle = (first + last) >>> 1;
             if (entries[middle].value.intValue() <= value
                     && ((middle + 1) >= entries.length || value < entries[middle + 1].value
                             .intValue())) {
